flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1254,9 +1291,9 @@ class ExprLowering {
   fir::ExtendedValue gen(const Fortran::evaluate::ArrayRef &aref) {
     if (aref.base().IsSymbol()) {
       auto &symbol = aref.base().GetFirstSymbol();
-      auto si = symMap.lookupSymbol(symbol);
```

Comment:
Was this a bug?

Reply:
Not really, it has to do with the fact that I did not create a SymbolBox classification for pointer/allocatable and used `SymbolBox::Intrinsic` since they are just a descriptor address in the symbol map until we create an ExtentdedValue from them. `genArrayCoorOp` is of course not made to work with a `SymbolBox::Intrinsic` base.
So I modified `genArrayCoorOp` to work with ExtendedValue instead of SymbolBox so and called `gen(symbol)` to get the descriptor unboxing done.
I thought it made some sense to allow `genArrayCoorOp` to operate on `ExtendedValue`, so I went for this short-cut, but I also think allocatble/pointers should not be stored as SymbolBox::Intrinsic in the long term (even if they just are a single mlir::Value too.)

Reply:
Agreed. We may want to expand the menu of extended values. Not all allocatables need a descriptor, etc.